lib/acts_as_taggable_on_mongoid/tag_list.rb

```diff
@@ -0,0 +1,169 @@
+# frozen_string_literal: true
+
+# require "active_support/core_ext/module/delegation"
+
+module ActsAsTaggableOnMongoid
+  # A list of tags.  The TagList must be initialized with a tag definition so that it knows how to clean
+  # the list properly and to convert the list to a string.
+  #
+  # All methods that add objects to the list (initialization, concat, etc.) optionally take an array of values including
+  # options to parse the values and to optionally specifiy the parser to use.  If no parser is specified, then
+  # parser for the tag_definition is used.
+  #
+  # If the input value(s) are to be parsed, then all values passed in are parsed.
+  #
+  # Examples:
+  #   TagList.new(tag_definition, "value 1", "value 2")
+  #   # > TagList<> ["value 1", "value 2"]
+  #
+  #   TagList.new(tag_definition, "value 1, value 2", parse: true)
+  #   # > TagList<> ["value 1", "value 2"]
+  #
+  #   TagList.new(tag_definition, "value 1, value 2", "value 3, value 4", parse: true)
+  #   # > TagList<> ["value 1", "value 2", "value 3", "value 4"]
+  #
+  #   TagList.new(tag_definition, "value 1, value 2", "value 3, value 4", parser: ActsAsTaggableOnMongoid::GenericParser)
+  #   # > TagList<> ["value 1", "value 2", "value 3", "value 4"]
+
+  # :reek:MissingSafeMethod
+  # :reek:SubclassedFromCoreClass
+  class TagList < Array
```

Comment:
Copied from here:
https://github.com/mbleigh/acts-as-taggable-on/blob/master/lib/acts_as_taggable_on/tag_list.rb

Most of the code was copied as-is.
Because I moved all configurations out of the Configurations into the TagTypeDefinition, added 2 primary things:
* tag_definition - to control actions such as downcase, parameterization, ets.
* taggable - if the list changes, and this is set then notify the owner class of changes to allow it to track changes.